library/core/src/ptr/mod.rs

```diff
@@ -2209,6 +2209,9 @@ impl<F: FnPtr> fmt::Debug for F {
 
 /// Creates a `const` raw pointer to a place, without creating an intermediate reference.
 ///
+/// `addr_of!(expr)` is equivalent to `&raw const expr`. The macro is *soft-deprecated*;
+/// use `&raw const` instead.
```

Comment:
Remark (not in scope of this PR): this probably wants an allow-by-default deprecation lint or something eventually that can be bumped up to a warning.

Reply:
We have "deprecated in a future version of Rust" that we can use for that, but I intended for this to be even softer to avoid too much churn.